plugin/FilterModule/MarkupParser.cs

```diff
@@ -41,6 +41,20 @@ public string TransformImgToPicture(string content)
                     if ((src != null && src.Value.IndexOf("slimmage=true", StringComparison.InvariantCultureIgnoreCase) > -1)
                         || (cls != null && cls.Value.IndexOf("slimmage",  StringComparison.InvariantCultureIgnoreCase) > -1))
                     {
+                        // - if no width set with querystring param, but preset name, get the width from the preset
+                        if ((src.Value.IndexOf("width=", StringComparison.InvariantCultureIgnoreCase) == -1)
+                            && (src.Value.IndexOf("preset=", StringComparison.InvariantCultureIgnoreCase) > -1))
+                        {
+                            // - get presets from config
+                            var configSection = WebConfigurationManager.GetSection("resizer") as ImageResizer.ResizerSection;
+                            XmlElement conf = configSection.getCopyOfNode("presets").ToXmlElement();
+                            // - find preset by name
+                            XmlNode curpreset = conf.SelectSingleNode(string.Format("preset[@name='{0}']/@defaults", System.Web.HttpUtility.ParseQueryString(src.Value.Substring(src.Value.IndexOf("?")))["preset"]));
+                            string presetWidth = System.Web.HttpUtility.ParseQueryString(curpreset.InnerText)["width"];
+
+                            // - append presets's default width to querystring
+                            src.Value = string.Format("{0}&width={1}", src.Value, presetWidth);
```

Comment:
Now that slimmage supports height, perhaps we can just apply the preset client-side as if it were server side? No item-specific logic? Perhaps check the preset xml itself for an enabling flag, like 'applyWithSlimResponse'?